### PR TITLE
Bug 1148347 - [Gallery] Check for edit view and close editor object when device is mounted for USB mass storage r=djf

### PR DESCRIPTION
**Pull request by punamdahiya**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a91f0873585ba739070c4bbaed9b62b80686c163

File: apps/gallery/js/startup.js [hunk 1]
```diff
@@ -93,6 +93,11 @@
       // If we were viewing or editing an image it might not be there
       // anymore when the MediaDB becomes available again.
       if (!picking) {
+        // If we are editing image, call exitEdit to revoke blob URLs
+        // and close the editor object.
```
Comment (davidflanagan) [inline apps/gallery/js/startup.js]: Please add a note indicating that this is a known case where the user can lose their work.

File: apps/gallery/js/ImageEditor.js [hunk 2]
```diff
@@ -644,13 +644,15 @@ function exitEdit(saved) {
...
-  // We came in to edit mode from fullscreenView.  If the user cancels the edit
-  // go back to fullscreenView.  Otherwise, if the user saves the photo, we go
-  // back to thumbnail list view because that is where the newly saved
-  // image is going to show up.
-  // XXX: this isn't really right. Ideally the new photo should show up
-  // right next to the old one and we should go back to fullscreenView to view
-  // the edited photo.
+  // If exitEdit is called without saved argument e.g. when device storage
+  // is unavailable, exit now and do not check for saved.
+  if (arguments.length < 1) {
```
Comment (davidflanagan) [inline apps/gallery/js/ImageEditor.js]: Using |arguments| is an instant performance hit for JavaScript. Instead, you could just say:

if (saved === undefined)

This assumes that you've verified that all existing calls to exitEdit actually pass a value.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=930f5839473550d594e23d592c6f8a75e8666ef6

File: apps/gallery/js/startup.js (hunk 1 above)
Comment (punamdahiya) [inline apps/gallery/js/startup.js]: Done

File: apps/gallery/js/ImageEditor.js (hunk 2 above)
Comment (punamdahiya) [inline apps/gallery/js/ImageEditor.js]: That's correct exitEdit passes values saved for the two cases inside ImageEditor.js

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=db214f275cadfdb6f51cab89e27f8a8fb5662de1

**Pull request merged**